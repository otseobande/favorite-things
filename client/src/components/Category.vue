<template>
  <div :class="`
    max-h-screen
    min-h-10
    overflow-auto
    m-0
    mr-4
    bg-gray-500
    rounded
    text-gray-100
    category`"
  >
    <div :class="`
      sticky
      bg-gray-500
      pt-4
      pl-4
      pr-3
      top-0
      m-0
      mb-3
      font-extrabold
      text-xl
      capitalize
      category-title`"
    >
      <div v-if="category.name" class="flex justify-between">
        <t-input
          v-if="edittingCategoryName"
          v-model="updatedCategory.name"
          class="h-8 bg-white border-white text-gray-800 w-full text-sm"
          @blur="updateCategoryName"
          @keyup.enter="$event.target.blur()"
          ref="updateCategoryNameInput"
          autofocus
        />
        <span
          v-else
          @click="handleCategoryNameClick"
        >
          {{category.name}}
        </span>

        <button class="p-2" @click="showConfirmDeleteModal">
          <CloseIcon class="fill-current w-4" />
        </button>
      </div>
      <div v-else>
        <label class="text-sm">Category name</label>
        <t-input
          v-model="newCategoryName"
          class="h-8 bg-white border-white text-gray-800 w-full text-sm"
          placeholder="Enter new category name"
          @blur="handleNewCategoryInputBlur"
          @keyup.enter="$event.target.blur()"
          ref="newCategoryInput"
          autofocus
        />
      </div>
    </div>
      <div class="w-full px-2">
        <draggable
          :list="category.favorite_things"
          group="categories"
          @start="drag=true"
          @end="drag=false"
          @change="handleFavoriteThingChange"
          v-bind="dragOptions"
        >
          <transition-group type="transition" :name="!drag ? 'flip-list' : null">
            <FavoriteThing
              v-for="(favoriteThing, index) in category.favorite_things"
              :key="favoriteThing.id"
              :favoriteThing="favoriteThing"
              :index="index"
              :categoryName="category.name"
              :category="category"
              :singularizedCategoryName="singularizedCategoryName"
            />
          </transition-group>
        </draggable>
      </div>
    <!-- </div> -->

    <div v-if="category.name" class="bg-gray-500 sticky bottom-0">
      <button
        class="bottom-0 outline-none mt-3 p-3 py-3 hover:bg-gray-600 cursor-pointer font-bold w-full px-2 py-1 rounded"
        @click="showNewFavoriteThingModal = true"
      >
        <img width="20" class="inline" src="../assets/add-icon-white.svg" />
        <span class="ml-2 font-bold">
          Add favorite <span class="lowercase">{{singularizedCategoryName}}</span>
        </span>
        <!-- Icons made by https://www.flaticon.com/authors/hanan from https://www.flaticon.com/ -->
      </button>
    </div>
    <Modal
      :show="showNewFavoriteThingModal"
      @close="showNewFavoriteThingModal = false"
    >
      <NewFavoriteThing
        :category="category"
        :singularizedCategoryName="singularizedCategoryName"
        @finished="showNewFavoriteThingModal = false"
      />
    </Modal>
    <t-modal ref="categoryDeleteModal" closeIconClass="text-gray-300 w-8 right-0 mt-2 mr-6 absolute" class="h-30">
      <div class="text-gray-700">
        <div class="text-lg mt-16 text-center">
           <h2 class="font-bold mb-4">Add you sure you want to delete {{category.name}}?</h2>
           <p>Deleting a category would delete all the favorite items in that category.</p>
        </div>
        <div class="mt-6 flex justify-center">
          <div>
            <t-button
              variant="danger"
              size="sm"
              class="mr-2 font-bold"
              type='submit'
              @click="deleteCategory"
            >
              <span>
                Delete
              </span>
            </t-button>
            <t-button size="sm" class="font-bold" @click="closeConfirmDeleteModal">
              Cancel
            </t-button>
          </div>
        </div>
      </div>
    </t-modal>
  </div>
</template>

<script>
// Vuex
import { mapActions } from 'vuex';
import * as actionTypes from '../store/actions.type';

import draggable from 'vuedraggable';
import Tense from 'tense';
import toast from 'toast-me';

// components
import FavoriteThing from './FavoriteThing.vue';
import NewFavoriteThing from './NewFavoriteThing.vue';
import Modal from './Modal.vue';
import CloseIcon from './CloseIcon.vue'

export default {
  props: ['category'],
  components: {
    FavoriteThing,
    draggable,
    NewFavoriteThing,
    Modal,
    CloseIcon
  },
  data() {
    return {
      showModal: false,
      showNewFavoriteThingModal: false,
      newFavoriteThing: {},
      dragOptions: {
        animation: 200,
        group: "description",
        disabled: false,
        ghostClass: "ghost"
      },
      drag: false,
      newCategoryName: '',
      updatedCategory: {
        name: this.category.name
      },
      edittingCategoryName: false
    }
  },
  methods: {
    ...mapActions([
      actionTypes.DELETE_DUMMY_CATEGORY,
      actionTypes.UPDATE_FAVORITE_THING,
      actionTypes.UPDATE_CATEGORY,
      actionTypes.ADD_CATEGORY,
      actionTypes.DELETE_CATEGORY
    ]),
    handleFavoriteThingChange(event) {
      if(event.moved) {
        this[actionTypes.UPDATE_FAVORITE_THING]({
          favoriteThingId: event.moved.element.id,
          updatedFavoriteThingData: {
            ranking: event.moved.newIndex + 1
          }
        })
      }

      if (event.added) {
        this[actionTypes.UPDATE_FAVORITE_THING]({
          favoriteThingId: event.added.element.id,
          updatedFavoriteThingData: {
            ranking: event.added.newIndex + 1,
            category: this.category.id
          }
        })
      }
    },
    handleCategoryNameClick () {
      this.edittingCategoryName = true;
      setTimeout(() => this.$refs.updateCategoryNameInput.focus(), 100);
    },
    async updateCategoryName() {
      if (this.category.name !== this.updatedCategory.name) {
        await this[actionTypes.UPDATE_CATEGORY]({
          categoryId: this.category.id,
          updatedCategoryData: {
            name: this.updatedCategory.name
          }
        });
      }

      this.edittingCategoryName = false;
    },
    async handleNewCategoryInputBlur() {
      try{
        if(this.newCategoryName) {
          await this[actionTypes.ADD_CATEGORY]({
            name: this.newCategoryName
          });
        }
      } catch (error) {
        if (error.response.status === 400) {
          if (error.response.data.name) {
            toast(error.response.data.name[0], 'error');
          }
        }
      } finally {
        this[actionTypes.DELETE_DUMMY_CATEGORY]();
      }
    },
    showConfirmDeleteModal() {
      this.$refs.categoryDeleteModal.show();
    },
    closeConfirmDeleteModal() {
      this.$refs.categoryDeleteModal.hide()
    },
    deleteCategory() {
      this.$refs.categoryDeleteModal.hide();
      return this[actionTypes.DELETE_CATEGORY]({
        categoryId: this.category.id
      });
    },
  },
  computed: {
    singularizedCategoryName() {
      const tense = new Tense()
      return tense.singularize(this.category.name)
    }
  },
  mounted() {
    if(this.$refs.newCategoryInput) {
      this.$refs.newCategoryInput.focus()
    }
  }
}
</script>

<style>
.category{
  max-height: 85vh;
  min-width: 262px;
}

.type-select > select {
  height: 2.5rem;
}
.flip-list-move {
  transition: transform 0.5s;
}

.no-move {
  transition: transform 0s;
}

.ghost {
  opacity: 0.5;
  background: #c8ebfb;
}

.list-group {
  min-height: 20px;
}

.list-group-item {
  cursor: move;
}

.list-group-item i {
  cursor: pointer;
}
</style>
