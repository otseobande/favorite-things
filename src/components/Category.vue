<template>
  <div class="min-h-2 max-h-full w-64 p-3 m-0 mr-4 bg-gray-500 rounded text-gray-100 font-bold">
    <h2 class="m-0 mb-3 ml-5 font-extrabold text-xl">Food</h2>
    <div class="flex">
      <div class="w-3 h-12 flex-col content-center justify-center" >
        <div
          class="h-full mt-2 align-middle"
          v-for="(favoriteThing, index) in favoriteThings"
          :key="favoriteThing.id">{{index + 1}}.</div>
      </div>
      <div class="w-10/12">
        <draggable
          v-model="favoriteThings"
          group="people"
          @start="drag=true"
          @end="drag=false"
          v-bind="dragOptions"
        >
          <transition-group type="transition" :name="!drag ? 'flip-list' : null">
            <FavoriteThing v-for="favoriteThing in favoriteThings" :key="favoriteThing.id" :favoriteThing="favoriteThing" />
          </transition-group>
        </draggable>
      </div>
    </div>

    <button class="outline-none mt-3 hover:bg-gray-600 cursor-pointer font-bold w-full px-2 py-1 rounded ">
      <img width="20" class="inline" src="../assets/add-icon-white.svg" />
      <span class="ml-2">Add favorite food</span>
      <!-- Icons made by https://www.flaticon.com/authors/hanan from https://www.flaticon.com/ -->
    </button>
  </div>
</template>

<script>
import FavoriteThing from './FavoriteThing.vue';
import draggable from 'vuedraggable';

export default {
  data() {
    return {
      favoriteThings: [
        {
          id: 1,
          name: 'Plantain'
        },
        {
          id: 2,
          name: 'Rice'
        },
        {
          id: 3,
          name: 'Beans'
        },
        {
          id: 4,
          name: 'Semo'
        }
      ],
      dragOptions: {
        animation: 200,
        group: "description",
        disabled: false,
        ghostClass: "ghost"
      },
      drag: false
    }
  },
  components: {
    FavoriteThing,
    draggable
  }
}
</script>

<style>
.flip-list-move {
  transition: transform 0.5s;
}
.no-move {
  transition: transform 0s;
}
.ghost {
  opacity: 0.5;
  background: #c8ebfb;
}
.list-group {
  min-height: 20px;
}
.list-group-item {
  cursor: move;
}
.list-group-item i {
  cursor: pointer;
}
</style>
