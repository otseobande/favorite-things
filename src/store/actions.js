import * as actionTypes from './actions.type';
import * as mutationTypes from './mutations.type';

import categoryService from '../services/categoryService';
import favoriteThingService from '../services/favoriteThingService';

export default {
  async [actionTypes.FETCH_CATEGORIES]({ commit }) {
    const response = await categoryService.get();

    commit(mutationTypes.SET_CATEGORIES, response.data.results)
  },

  async [actionTypes.ADD_FAVORITE_THING]({ commit }, newFavoriteThing) {
    const response = await favoriteThingService.create(newFavoriteThing);

    commit(mutationTypes.ADD_FAVORITE_THING, response.data)
  },

  async [actionTypes.UPDATE_FAVORITE_THING]({ commit }, {
    favoriteThingId,
    updatedFavoriteThingData
  }) {
    const response = await favoriteThingService.update({
      favoriteThingId,
      updatedFavoriteThingData
    })

    commit(mutationTypes.UPDATE_FAVORITE_THING, response.data);
  },

  async [actionTypes.DELETE_FAVORITE_THING]({ commit }, { favoriteThingId }) {
    await favoriteThingService.delete({
      favoriteThingId
    });

    commit(mutationTypes.DELETE_FAVORITE_THING, favoriteThingId)
  },

  async [actionTypes.ADD_DUMMY_CATEGORY]({ commit }) {
    commit(mutationTypes.ADD_CATEGORY, {
      id: 'template',
      name: '',
      favorite_things: []
    })
  },

  async [actionTypes.DELETE_DUMMY_CATEGORY]({ commit }) {
    commit(mutationTypes.DELETE_CATEGORY, 'template')
  },

  async [actionTypes.ADD_CATEGORY]({ commit }, newCategory) {
    const response = await categoryService.create(newCategory)

    commit(mutationTypes.ADD_CATEGORY, response.data)
  },

  async [actionTypes.DELETE_CATEGORY]({ commit }, { categoryId }) {
    await categoryService.delete(categoryId);

    commit(mutationTypes.DELETE_CATEGORY, categoryId);
  }
}
