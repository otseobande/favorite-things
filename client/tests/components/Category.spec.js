import { shallowMount, createLocalVue } from '@vue/test-utils';
import Category from '../../src/components/Category.vue';
import VueTailwind from 'vue-tailwind';
import Vuex from 'vuex';
import actions from '../../src/store/actions';
import * as actionTypes from '../../src/store/actions.type';
import toast from 'toast-me';

jest.mock('../../src/store/actions');
jest.mock('toast-me');
jest.useFakeTimers();

const localVue = createLocalVue();

localVue.use(VueTailwind);
localVue.use(Vuex);
const stubs = {
  't-input': {
    render() {},
    methods: {
      focus: jest.fn()
    }
  }
}
let wrapper;
const propsData = {
  category: {
    name: 'Test category',
    favorite_things: [
      {
        title: 'Test thing',
        ranking: 1,
        category: 1,
        metadata: [],
        history: [
          {
            action: 1,
            changes: {
              'name': ['People', 'Places'],
              'title': ['food', 'sport'],
              'job': ['cleaner', 'software developer']
            }
          },
          {
            action: 2,
            changes: {
              'name': ['People', 'Places'],
              'title': ['food', 'sport'],
              'job': ['cleaner', 'software developer']
            }
          }
        ]
      }
    ]
  },
}

describe('Category component', () => {
  beforeEach(() => {
    wrapper = shallowMount(Category, {
      localVue,
      store: new Vuex.Store({
        actions
      }),
      propsData,
      stubs
    });
  })
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleFavoriteThingChange method', () => {
    it('should handle move event', () => {
      wrapper.vm[actionTypes.UPDATE_FAVORITE_THING] = jest.fn();

      const event = {
        moved: {
          newIndex: 1,
          element: {
            id: 1
          }
        }
      }

      wrapper.vm.handleFavoriteThingChange(event);

      expect(wrapper.vm[actionTypes.UPDATE_FAVORITE_THING]).toBeCalledWith({
        favoriteThingId: event.moved.element.id,
        updatedFavoriteThingData: {
          ranking: event.moved.newIndex + 1
        }
      })
    });

    it('should handle added event', () => {
      wrapper.vm[actionTypes.UPDATE_FAVORITE_THING] = jest.fn();

      const event = {
        added: {
          newIndex: 1,
          element: {
            id: 1
          }
        }
      }

      wrapper.vm.handleFavoriteThingChange(event);

      expect(wrapper.vm[actionTypes.UPDATE_FAVORITE_THING]).toBeCalledWith({
        favoriteThingId: event.added.element.id,
        updatedFavoriteThingData: {
          category: wrapper.vm.category.id,
          ranking: event.added.newIndex + 1,
        }
      })
    })
  });

  describe('handleCategoryNameClick method', () => {
    it('should set editting to true', () => {
      wrapper.vm.$refs.updateCategoryNameInput = {
        focus: jest.fn()
      }
      wrapper.vm.handleCategoryNameClick()

      expect(wrapper.vm.edittingCategoryName).toBeTruthy();
      expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 100);

      setTimeout.mock.calls[0][0]()

      expect(stubs['t-input'].methods.focus).toBeCalled()
      stubs['t-input'].methods.focus.mockReset();
    })
  });

  describe('updateCategoryName', () => {
    it('should set edittingCategoryName to false', async () => {
      await wrapper.vm.updateCategoryName()

      expect(wrapper.vm.edittingCategoryName).toBe(false);
    });

    it('should update category if name is changed', async () => {
      wrapper.vm[actionTypes.UPDATE_CATEGORY] = jest.fn();
      wrapper.vm.updatedCategory = {
        name: 'Another name'
      }

      await wrapper.vm.updateCategoryName();

      expect(wrapper.vm[actionTypes.UPDATE_CATEGORY]).toBeCalledWith({
        categoryId: wrapper.vm.category.id,
        updatedCategoryData: {
          name: wrapper.vm.updatedCategory.name
        }
      });
      expect(wrapper.vm.edittingCategoryName).toBe(false);
    })
  });

  describe('handleNewCategoryInputBlur', () => {
    it('should call DELETE_DUMMY_CATEGORY if newCategoryName is empty', async () => {
      wrapper.vm[actionTypes.DELETE_DUMMY_CATEGORY] = jest.fn()

      wrapper.vm.handleNewCategoryInputBlur()

      expect(wrapper.vm[actionTypes.DELETE_DUMMY_CATEGORY]).toBeCalled()
    });

    it('should add new category if category name is set', async () => {
      wrapper.vm[actionTypes.DELETE_DUMMY_CATEGORY] = jest.fn();
      wrapper.vm[actionTypes.ADD_CATEGORY] = jest.fn();
      wrapper.vm.newCategoryName = 'new category';

      wrapper.vm.handleNewCategoryInputBlur()

      expect(wrapper.vm[actionTypes.ADD_CATEGORY]).toBeCalledWith({
        name: wrapper.vm.newCategoryName
      })
    })

    it('should toast error if add category returns error', async () => {
      wrapper.vm[actionTypes.DELETE_DUMMY_CATEGORY] = jest.fn();
      const error = {
        response: {
          status: 400,
          data: {
            name: ['Name is required']
          }
        }
      }

      wrapper.vm[actionTypes.ADD_CATEGORY] = jest.fn(() => {
        return Promise.reject(error)
      });
      wrapper.vm.newCategoryName = 'new category';

      await wrapper.vm.handleNewCategoryInputBlur()

      expect(toast).toBeCalledWith(error.response.data.name[0], 'error')
      toast.mockReset()
    })

    it('should not toast error if add category returns error not 400', async () => {
      wrapper.vm[actionTypes.DELETE_DUMMY_CATEGORY] = jest.fn();
      const error = {
        response: {
          status: 301,
        }
      }

      wrapper.vm[actionTypes.ADD_CATEGORY] = jest.fn(() => {
        return Promise.reject(error)
      });
      wrapper.vm.newCategoryName = 'new category';

      await wrapper.vm.handleNewCategoryInputBlur()

      expect(toast).not.toBeCalled()
      toast.mockReset()
    })
  })

  describe('showConfirmDeleteModal', () => {
    it('should show confirm delete modal', () => {
      wrapper.vm.$refs.categoryDeleteModal = {
        show: jest.fn()
      }

      wrapper.vm.showConfirmDeleteModal()

      expect(wrapper.vm.$refs.categoryDeleteModal.show).toBeCalled()
    })
  })

  describe('closeConfirmDeleteModal', () => {
    it('should close confirm delete modal', () => {
      wrapper.vm.$refs.categoryDeleteModal = {
        hide: jest.fn()
      }

      wrapper.vm.closeConfirmDeleteModal()

      expect(wrapper.vm.$refs.categoryDeleteModal.hide).toBeCalled()
    })
  })

  describe('deleteCategory', () => {
    it('should hide delete modal and call delete action', () => {
      wrapper.vm[actionTypes.DELETE_CATEGORY] = jest.fn();
      wrapper.vm.$refs.categoryDeleteModal = {
        hide: jest.fn()
      }

      wrapper.vm.deleteCategory()

      expect(wrapper.vm.$refs.categoryDeleteModal.hide).toBeCalled()
      expect(wrapper.vm[actionTypes.DELETE_CATEGORY]).toBeCalledWith({
        categoryId: wrapper.vm.category.id
      })
    })
  })

  describe('mounted', () => {
    it('should focus input', () => {
      wrapper = shallowMount(Category, {
        localVue,
        store: new Vuex.Store({
          actions
        }),
        propsData: {
          ...propsData,
          category: {
            ...propsData.category,
            name: ''
          }
        },
        stubs
      });

      expect(stubs['t-input'].methods.focus).toBeCalled()
    })
  })
});
