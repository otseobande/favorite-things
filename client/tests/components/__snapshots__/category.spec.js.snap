// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category component should match snapshot 1`] = `
<div class="
  max-h-screen
  min-h-10
  overflow-auto
  m-0
  mr-4
  bg-gray-500
  rounded
  text-gray-100
  category">
  <div class="
    sticky
    bg-gray-500
    pt-4
    pl-4
    pr-3
    top-0
    m-0
    mb-3
    font-extrabold
    text-xl
    capitalize
    category-title">
    <div class="flex justify-between"><span>
        Test category
      </span> <button class="p-2">
        <closeicon-stub class="fill-current w-4"></closeicon-stub>
      </button></div>
  </div>
  <div class="w-full px-2">
    <draggable-stub list="[object Object]" clone="function(t){return t}" element="div" group="categories" animation="200" ghostclass="ghost"><span type="transition" name="flip-list"><favoritething-stub favoritething="[object Object]" index="0" category="[object Object]" singularizedcategoryname="Test category" categoryname="Test category"></favoritething-stub></span></draggable-stub>
  </div>
  <div class="bg-gray-500 sticky bottom-0"><button class="bottom-0 outline-none mt-3 p-3 py-3 hover:bg-gray-600 cursor-pointer font-bold w-full px-2 py-1 rounded"><img width="20" src="../assets/add-icon-white.svg" class="inline"> <span class="ml-2 font-bold">
        Add favorite <span class="lowercase">Test category</span></span></button></div>
  <modal-stub>
    <newfavoritething-stub category="[object Object]" singularizedcategoryname="Test category"></newfavoritething-stub>
  </modal-stub>
  <t-modal-stub clicktoclose="true" esctoclose="true" width="600" height="300" pivoty="0.1" baseclass="z-40 overflow-auto left-0 top-0 bottom-0 right-0 w-full h-full fixed" overlayclass="z-40 overflow-auto left-0 top-0 bottom-0 right-0 w-full h-full fixed bg-black opacity-75" containerclass="z-50 relative p-3 mx-auto my-0 max-w-full" wrapperclass="bg-white rounded shadow-lg border flex flex-col overflow-hidden" bodyclass="p-6 flex-grow" headerclass="px-6 py-3 text-xl border-b" footerclass="px-6 py-3 border-t" closeiconclass="text-gray-300 w-8 right-0 mt-2 mr-6 absolute" class="h-30">
    <div class="text-gray-700">
      <div class="text-lg mt-16 text-center">
        <h2 class="font-bold mb-4">Add you sure you want to delete Test category?</h2>
        <p>Deleting a category would delete all the favorite items in that category.</p>
      </div>
      <div class="mt-6 flex justify-center">
        <div>
          <t-button-stub tagname="button" type="submit" variant="danger" size="sm" activeclass="router-link-active" exactactiveclass="router-link-exact-active" baseclass="border block rounded inline-flex items-center justify-center" defaultclass="bg-white border-gray-400 hover:bg-gray-100 hover:border-gray-500" primaryclass="text-white bg-blue-500 border-blue-500 hover:bg-blue-600 hover:border-blue-600" secondaryclass="border-blue-500 text-blue-500 bg-white hover:border-blue-600 hover:text-blue-600 hover:bg-white" tertiaryclass="border block underline text-blue-500 border-transparent bg-transparent hover:text-blue-600" successclass="text-white bg-green-500 border-green-500 hover:bg-green-600 hover:border-green-600" dangerclass="text-white bg-red-500 border-red-500 hover:bg-red-600 hover:border-red-600" warningclass="text-yellow-900 bg-yellow-500 border-yellow-500 hover:bg-yellow-600 hover:border-yellow-600" disabledclass="cursor-not-allowed opacity-75" defaultsizeclass="px-6 py-3" largesizeclass="px-8 py-4 text-lg" smallsizeclass="px-4 py-2 text-sm" class="mr-2 font-bold"><span>
              Delete
            </span></t-button-stub>
          <t-button-stub tagname="button" type="button" size="sm" activeclass="router-link-active" exactactiveclass="router-link-exact-active" baseclass="border block rounded inline-flex items-center justify-center" defaultclass="bg-white border-gray-400 hover:bg-gray-100 hover:border-gray-500" primaryclass="text-white bg-blue-500 border-blue-500 hover:bg-blue-600 hover:border-blue-600" secondaryclass="border-blue-500 text-blue-500 bg-white hover:border-blue-600 hover:text-blue-600 hover:bg-white" tertiaryclass="border block underline text-blue-500 border-transparent bg-transparent hover:text-blue-600" successclass="text-white bg-green-500 border-green-500 hover:bg-green-600 hover:border-green-600" dangerclass="text-white bg-red-500 border-red-500 hover:bg-red-600 hover:border-red-600" warningclass="text-yellow-900 bg-yellow-500 border-yellow-500 hover:bg-yellow-600 hover:border-yellow-600" disabledclass="cursor-not-allowed opacity-75" defaultsizeclass="px-6 py-3" largesizeclass="px-8 py-4 text-lg" smallsizeclass="px-4 py-2 text-sm" class="font-bold">
            Cancel
          </t-button-stub>
        </div>
      </div>
    </div>
  </t-modal-stub>
</div>
`;
